shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -252,7 +252,8 @@ public void notifyLowMemoryWarning() {
   public static class DartEntrypoint {
     @NonNull
     public static DartEntrypoint createDefault() {
-      return new DartEntrypoint(FlutterMain.findAppBundlePath(), "main");
+      return new DartEntrypoint(
+          FlutterInjector.instance().flutterLoader().findAppBundlePath(), "main");
```

Comment:
This still implies findAppBundlePath needs to return a default if FlutterLoader is not inited.

Reply:
This is somewhat explicit but the DartExecutor is created by the engine and only creates it after having loaded the native library. If I understood your question correctly. We can add asserts and stuff too to make the dependency explicit.

Reply:
Yes, we should add asserts since this is public API IIRC.

Reply:
Turns out this needs a sizable refactor 🤣 https://github.com/flutter/engine/pull/20789/commits/3ce3419fddb1f8dc756ffe6cea4c285a5f519a3c

PTAL